### PR TITLE
fix(open-pr-comments): Update Regex for Snuba Query Error

### PR DESCRIPTION
**Pull request by iamrajjoshi**
* I updated the regex query so when looking for a function name during arrow functions, we make sure we don't grab other tokens from the patch that exist before the declaration by ignoring matches that include an end line after a const declaration (and before any function keyword)
* Fixed unrelated regex bug with the ignore list to match how python regex syntax
* Wrote tests

Fixes:  https://github.com/getsentry/team-core-product-foundations/issues/162

### CODE REVIEW

File: tests/sentry/tasks/integrations/github/test_language_parsers.py [hunk 1]
```diff
@@ -578,6 +578,50 @@ def test_typescript_example(self):
...
         }
 
+    def test_javascript_functions_after_const(self):
+
+        patch = """
+          "@@ -305,9 +285,7 @@ export const Redacted Redacted\n   // Redacted.\n   // Redacted \n   const redacted = redacted;\n+  const redacted = redacted();\n+  // const redacted = true;\n+  const redacted = redacted()\n+    redacted; // Redacted\nconst \n@@ -165,24 +171,40 @@ export const RedactedRedactedRedactedRedactedRedacted
+
+          // Redacted
+@@44,38@@ var arrow2 = (a, ...b) => 0;
+    """
```
Comment (cathteng) [inline tests/sentry/tasks/integrations/github/test_language_parsers.py:588]: if you combine all of these patches does it work too?

Comment (iamrajjoshi) [reply tests/sentry/tasks/integrations/github/test_language_parsers.py:588]: i just combined the locally, and it does! should I combine them?

Comment (cathteng) [reply tests/sentry/tasks/integrations/github/test_language_parsers.py:588]: why not

Comment (iamrajjoshi) [reply tests/sentry/tasks/integrations/github/test_language_parsers.py]: done

**Review by cathteng — APPROVED** [review]
🚢

**Pull request merged**